Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/AbstractIntegerDataType.java

```diff
@@ -275,8 +279,19 @@ else if (bytes.length < nominalLen) {
 				System.arraycopy(bytes, 0, chars, nominalLen - bytes.length, bytes.length);
 				bytes = chars;
 			}
+
 			MemBuffer memBuf = new ByteMemBufferImpl(null, bytes, true);
-			return new StringDataInstance(this, settings, memBuf, charSize).getCharRepresentation();
+			StringDataInstance instance = new StringDataInstance(this, settings, memBuf, nominalLen);
+			if (bytes.length == 1) {
+				return instance.getCharRepresentation();
+			}
+
+			String stringRepresentation = instance.getStringRepresentation();
+			if (stringRepresentation.length() != nominalLen) {
```

Comment:
Since most values returned from getStringRep() will have starting and end quotes, doesn't this check fail most of the time?
In which cases does the value returned from getStringRep() not meet your needs for the display?

Reply:
Character sequences beginning with \x00 (and other non-printable characters) for instance.
For those, the "Convert" context menu will display a quoted string like "\x00ABC" but getStringRep() will return an empty, quoted string.
StringUtilities.toQuotedString, on the other hand, is the same conversion function used by the context menu and therefore will always match the user's expected output.

That said, maybe it makes more sense to use StringUtilities.toQuotedString all the time then?

Reply:
StringDataInstance is the new hotness, StringUtilities.toQuotedString is yesterday.

StringDataInstance will format bad characters according to the RenderUnicodeSettingsDefinition, and ESC_SEQ should match up with StringUtilties.toQuotedString pretty good.

Maybe a new constructor on StringDataInstance that allows specifying the renderSetting directly instead of fetching it from the settings instance.

Reply:
This should be sorted out in the next commit, however the conversion context menu ought to use StringDataInstance too.  Unfortunately with these changes the listing won't match the context menu (\x00 vs \0, for example).